src/Heightmap.cc

```diff
@@ -324,11 +325,14 @@ Errors Heightmap::Load(ElementPtr _sdf, const ParserConfig &_config)
     return errors;
   }
 
+  auto config = _config;
+  auto path = std::filesystem::path(this->dataPtr->filePath).parent_path();
+  config.SetRelativeURISearchPath(path.string());
```

Comment:
I see that `SetRelativeURISearchPath` is called here to unconditionally override the previous value in the `ParserConfig` object passed to `Load`. I think that probably makes sense, I'm trying to think it through to be sure.

Do we want to emit a warning if the value isn't empty before overwriting it? I haven't had a chance to think this through yet.

cc @azeey

Reply:
I like @azeey's suggestion in https://github.com/gazebosim/sdformat/pull/1373#discussion_r1505134436